src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/FormattingPassBase.cs

```diff
@@ -589,7 +589,8 @@ protected bool ShouldFormat(FormattingContext context, TextSpan mappingSpan, boo
             }
 
             if (IsInHtmlTag() ||
-                IsInSingleLineDirective() ||
+                IsInDirectiveWithNoKind() ||
+                (!allowImplicitStatements && IsInSingleLineDirective()) ||
```

Comment:
Would you mind clarifying why the `allowImplicitStatements` bit is important here now?

Reply:
So this method is a little confusingly named (and my next PR might just be refactoring), but this method is answering the question "Should the C# formatter look at this?". Previously for any single line directive the answer was no. This change makes the answer yes if we're formatting implicit statements, or in other words, if we're formatting things that aren't blocks. Or at least in practice thats how it works.

This change is really only here for `@attribute` because its the only directive where there is a chunk of C# that is more than a single identifier.

Reply:
Ended up changing this, because it broke HTML formatting, and have made this specific to `@attribute` for now.

Logged https://github.com/dotnet/aspnetcore/issues/32419 for the real fix as a follow up